File: app-flutter/lib/pages/flutter/ui/tab/TabBar.dart
import 'package:flutter_web/material.dart';
// ======================================================================================================== 全局配置
import 'package:flutter_web_study.app/config.dart';
// ======================================================================================================== Widget 控件
class TabBarPage extends StatefulWidget { // 有状态的控件，控件的状态、文字等属性存在变化的可能
    final String title;
    TabBarPage({ Key key, this.title }) : super(key: key);
    @override
    _PageState createState() => _PageState(); // === createState()  => _PageState();
}
// SingleTickerProviderStateMixin
class _PageState extends State<TabBarPage> with TickerProviderStateMixin{
    // page
    @override
    Widget build(BuildContext context) {
        return Scaffold(
            appBar: Config.appBar(widget.title),
            body: ListView(
                children: <Widget>[
                    Container(
                        width: double.infinity,
                        color: Colors.white,
                        child: TabBar(
                            isScrollable: false,
                            indicatorColor: Colors.red,
                            indicatorSize: TabBarIndicatorSize.label,
                            indicatorWeight: 3.0,
                            labelPadding: EdgeInsets.all(10.0),
                            labelColor: Colors.red,
                            unselectedLabelColor: Color(0xffaaaaaa),
                            tabs: <Widget>[
                                Tab(text: '标题一',icon: Icon(Icons.tab),),
                                Tab(text: '标题二',icon: Icon(Icons.edit)),
                                Text('自定义')
                            ],
                            controller: TabController(vsync: this, length: 4),
                        ),
                    ),
                    Container(
                        margin: EdgeInsets.only(top:10.0),
                        width: double.infinity,
                        color: Colors.yellow,
                        child: TabBar(
                            isScrollable: true,
                            indicatorColor: Colors.red,
                            indicatorSize: TabBarIndicatorSize.tab,
                            labelColor: Colors.black,
                            tabs: <Widget>[
                                Tab(text: '标题一',icon: Icon(Icons.tab),),
                                Tab(text: '标题二',icon: Icon(Icons.edit)),
                                Tab(text: '标题三',icon: Icon(Icons.announcement)),
                                Tab(text: '标题四',icon: Icon(Icons.verified_user),),
                                Tab(text: '标题五',icon: Icon(Icons.satellite)),
                                Tab(text: '标题六',icon: Icon(Icons.mail)),
                            ],
                            controller: TabController(vsync: this, length: 6),
                        ),
                    )
                ],
            )
        );
    }
}

File: app-react/src/pages/flutter/ui/tab/TabBar.js
import React from 'react'
// ===================================================================== public component
import ListComponent from '@cpt/list.component'
// ===================================================================== component
const data = {
	title:'tab 切换栏',
	keyword:'TabBar( )',
	type:'TabBar',
	url:'TabBarPage',
	data:[
		{ 
			keyword:'tabs', type:'List<Widget>', pro:'选项按钮', 
			code:[{path:'flutter/ui/tab/TabBar/tabs'}],
			link:[{title:'Tab', url:'/flutter/ui/Tab'}]
		},
		{ keyword:'controller', type:'TabController', pro:'控制器', link:'/flutter/style/TabController' },
		{ keyword:'isScrollable', type:'bool', pro:'是否可以滚动，false：不滚动，tab 自动铺满' },
		{ keyword:'preferredSize', type:'Size', pro:'首先尺寸', link:'/flutter/style/Size' },
		{ keyword:'dragStartBehavior', type:'DragStartBehavior', pro:'拖动行为', code:[{ path:'flutter/public/DragStartBehavior'}] },
		
		{ deviceTitle:'label', keyword:'labelColor', type:'Color', pro:'选中的颜色', link:'/flutter/style/Colors' },
		{ keyword:'unselectedLabelColor', type:'Color', pro:'未先择的颜色', link:'/flutter/style/Colors' },
		{ keyword:'labelStyle', type:'TextStyle', pro:'选中的样式', link:'/flutter/style/TextStyle' },
		{ keyword:'unselectedLabelStyle', type:'TextStyle', pro:'未先择的样式', link:'/flutter/style/TextStyle' },
		{ keyword:'labelPadding', type:'EdgeInsetsGeometry', pro:'补白', link:'/flutter/style/offset/EdgeInsets' },
		
		{ deviceTitle:'指示线', keyword:'indicatorColor', type:'Color', pro:'颜色', link:'/flutter/style/Colors' },
		{ keyword:'indicatorWeight', type:'double', pro:'粗细' },
		{ keyword:'indicatorSize', type:'TabBarIndicatorSize', pro:'尺寸', code:[{path:'flutter/ui/tab/TabBar/TabBarIndicatorSize'}] },
		{ keyword:'indicatorPadding', type:'EdgeInsetsGeometry', pro:'补白', link:'/flutter/style/offset/EdgeInsets' },
		
		{ deviceTitle:'事件', keyword:'onTap', type:'void Function( int )', pro:'点击时执行', code:[{path:'flutter/ui/tab/onTap'}] },
	]
}
export default class Index extends React.Component{ render(){ return <ListComponent data={data} {...this.props}/> } }